### PR TITLE
arch-riscv: Add vector index load/store insns & fix bugs

### PR DESCRIPTION
**Pull request by huxuan0307**

### CODE REVIEW

**Comment by ksco** [comment]
@jerinjoy @XDUFanYang 
Hi guys, please review this PR, and if no one has comments for more than 24 hours, I will merge this.

File: src/arch/riscv/isa/decoder.isa [hunk 1]
```diff
@@ -2240,9 +2240,13 @@ decode QUADRANT default Unknown::unknown() {
...
                 // VWFUNARY0
                 0x10: decode VS1 {
-                    0x00: VectorNonSplitFormat::vfmv_f_s({{
-                        Fd_bits = Vs1_vu[0];
-                    }}, OPFVV, VectorDummyOp);
+                    0x00: decode VM {
+                        // The encodings corresponding to the masked versions
+                        // (vm=0) of vfmv.f.s are reserved
+                        0x1: VectorNonSplitFormat::vfmv_f_s({{
+                            Fd_bits = Vs1_vu[0];
```
Comment (jerinjoy) [inline src/arch/riscv/isa/decoder.isa:2327]: Unrelated to this change but still a bug - this should be `Vs2`.

Comment (XDUFanYang) [reply src/arch/riscv/isa/decoder.isa:2327]: I agree with you.

Comment (XDUFanYang) [reply src/arch/riscv/isa/decoder.isa:2327]: > Unrelated to this change but still a bug - this should be `Vs2`.

I agree with you.

Comment (jerinjoy) [reply src/arch/riscv/isa/decoder.isa:2327]: I have a fix here:
https://github.com/rivosinc/plct-gem5/commit/a4cfc00006a39c7c450591054f2bfbbe5b812850

We can leave this code as is and I'll include the fix in my next PR after this one is merged.

**Comment by jerinjoy** [comment]
[arch-riscv: Add vsse*.v](https://github.com/plctlab/plct-gem5/pull/12/commits/e8a0bf1f306464bf20508cbfe70638e348fce6b6)

This commit doesn't have any changes.

**Comment by huxuan0307** [comment]
> [arch-riscv: Add vsse*.v](https://github.com/plctlab/plct-gem5/pull/12/commits/e8a0bf1f306464bf20508cbfe70638e348fce6b6)
> 
> This commit doesn't have any changes.

Ok!, I removed it

**Pull request merged**